tunnel/default.go

```diff
@@ -286,6 +334,74 @@ func (t *tun) listen(link *link) {
 	}
 }
 
+// keepalive periodically sends keepalive messages to link
+func (t *tun) keepalive(link *link) {
+	keepalive := time.NewTicker(KeepAliveTime)
+	defer keepalive.Stop()
+
+	for {
+		select {
+		case <-t.closed:
+			return
+		case <-keepalive.C:
+			// send keepalive message
+			log.Debugf("Tunnel sending keepalive to link: %v", link.Remote())
+			if err := link.Send(&transport.Message{
+				Header: map[string]string{
+					"Micro-Tunnel":       "keepalive",
+					"Micro-Tunnel-Token": t.token,
+				},
+			}); err != nil {
+				log.Debugf("Error sending keepalive to link %v: %v", link.Remote(), err)
+				if err == io.EOF {
+					t.Lock()
+					delete(t.links, link.Remote())
+					t.Unlock()
+					return
+				}
+				// TODO: handle this error
+			}
+		}
+	}
+}
+
+// nodeLink attempts to connect to node and returns link if successful
+// it returns error if the link failed to be established
+func (t *tun) nodeLink(node string) (*link, error) {
```

Comment:
I was confused by the name nodeLink, I assumed it returned a link and was going to say to change it to getLink but then on further examination of the method it looks like it does the whole connect flow. Maybe it should be called setupLink? or just setup?

Reply:
yeah I called it `nodeLink` because it creates links to nodes, but come to think of it, `setupLink` is much better. Will change it.